strawberry/schema/schema_converter.py

```diff
@@ -135,6 +140,46 @@ def from_directive(self, directive: StrawberryDirective) -> GraphQLDirective:
             description=directive.description,
         )
 
+    def from_schema_directive(
+        self, directive: StrawberrySchemaDirective
+    ) -> GraphQLDirective:
+        cls = directive.wrap
+        module = sys.modules[cls.__module__]
+
+        args: Dict[str, GraphQLArgument] = {}
+        for field in dataclasses.fields(directive.wrap):
+            if is_private(field.type):
+                continue
+
+            if isinstance(field, StrawberryField):
+                default = field.default
+            else:
+                default = getattr(cls, field.name, UNSET)
+
+            if default == dataclasses.MISSING:
+                default = UNSET
+
+            arg = StrawberryArgument(
+                python_name=field.name,
+                graphql_name=None,
+                type_annotation=StrawberryAnnotation(
+                    annotation=field.type,
+                    namespace=module.__dict__,
+                ),
+                default=default,
+            )
+            args[self.config.name_converter.from_argument(arg)] = self.from_argument(
+                arg
+            )
+
```

Comment:
don't we do something like this somewhere else already?

Reply:
Hrm, not that I found (but I'm still no expert in strawberry's code).

This is similar to the `from_directive` code.